### DISCUSSION TITLE
2007/04/03/%E5%A6%82%E4%BD%95%E8%B0%83%E8%AF%95-labview-%E8%B0%83%E7%94%A8%E7%9A%84dll

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#149] [Announcements]
# 2007/04/03/%E5%A6%82%E4%BD%95%E8%B0%83%E8%AF%95-labview-%E8%B0%83%E7%94%A8%E7%9A%84dll

https://labview.qizhen.xyz/

https://blog.qizhen.xyz/2007/04/03/%E5%A6%82%E4%BD%95%E8%B0%83%E8%AF%95-labview-%E8%B0%83%E7%94%A8%E7%9A%84dll

**Comment by ruanqizhen** [comment]
之前博客下的留言：

Qizhen
2009年03月11日 17:25 编辑
这个问题很具体，没没法调试也是找不到问题的。你最好找周围熟悉C语言的人帮忙看一下。

回复
梦蝶
2009年03月3日 20:36 编辑
能帮我看看这段数据采集卡的驱动有没有什么问题啊为什么我编译不了啊#include <windows.h>#include <dos.h>#define DLLExport_declspec(dllexport)DLLExport float hydaq (int chreg){ float value;int i,a,c,base,dl,dh;base=0x280;a=inportb(base+4);outportb(base,0);outportb(base+1,chreg);for(i=0;i<70;i++)outportb(base+2,0);do{c=inportb(base);}while(!(c&0x40));dl=inportb(base+3);dh=inportb(base+4)&0xf;value=((dh*256+dl)-2048)*10/4096;return value;}

回复
Qizhen
2008年05月27日 10:03 编辑
天鹰，你这恐怕是很细致的某一处错误导致的问题，恐怕不是我一下就能说到问题所在的。一般这样常见的问题就是参数配置有误，你还没有理解他的函数每个参数的意义，应当如何设置。建议和厂家或做过类似工作的人联系一下，问他们应当如何配置。

回复
新刚
2008年05月26日 17:38 编辑
长侠:
   你好，我看了你的博客，关于labview调用dll的文章，我最近就在调试，我用的是中泰pci8333采集卡，外部使用数字信号发生器产生两路正旋波，用8333来采集，其中就是用到了cln调用8333的动态库pci8kp.dll，但调试的返回值均为0，表明驱动起来了，但就是采不回来数，我在想是不是我的调用语句使用得有问题？还请长侠多多指教！
      
                                                                                                  天鹰
 

回复
Tom Lu
2007年04月6日 08:42 编辑
我一般是用MessageBox将传进来的参数显示出来看对不对, 虽然很傻的方法，但很实用。

回复
arrowpig
2007年04月4日 12:16 编辑
首先要确定调用规则是不是设对了,LabVIEW里面的配置一定要和被调函数一致.__cdecl还是__stdcall，这两者的压栈顺序一样，从右向左，但是负责清栈不一样。其次要确定LabVIEW里设定的参数个数，类型和被调函数一致。LabVIEW是动态调用DLL，不会在编译时替你检查参数的个数和类型。运行时只是完全按照你的配置压栈，退栈。文件打开失败一般有两个原因，文件名字不对，或者权限有问题。首先debug进你的Dll，确定文件名是不是被正确传递了。如果传递不正确，试着确认一下LabVIEW CLN中的配置，例如，C的string用CSTR。如果文件名字是对的，确定文件I/O操作的返回值。LabVIEW有个特点，LabVIEW的VI和LabVIEW本身是在同一个进程中运行的，也就是说就算VI停止运行，你的DLL还是在内存中，看看是不是你的文件正在被别人打开，而你又试图以独占方式打开文件。
关于调试，Debug和Release的区别在于Debug模式下编译器会试图加入一些附加信息以给调试者更多的提示，设置_DEBUG宏，assert函数生效。Release模式下设置编译器产生调试信息后，在调试时和Debug没有本质区别。在MSVC中可以由DLL工程F5加载LabVIEW进程，或者在Debug菜单下以Attach Process加载。

回复
